### ISSUE TITLE
Windows Jenkins builds are failing to missing Deterministic parameter on Vbc/Csc task

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#6211]
[Log](http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit64/379/consoleFull#-189014258379494335-f7bd-47d0-8771-8661e00c2db2)

```
20:49:47 d:\j\workspace\roslyn_prtest_win_dbg_unit64>msbuild /nologo /v:m /m /p:BuildVersion=0.0.0.0 d:\j\workspace\roslyn_prtest_win_dbg_unit64\build/Toolset.sln /p:NuGetRestorePackages=false /p:Configuration=Debug 
20:49:51 
C:\Users\dotnet-bot\.nuget\packages\Microsoft.Net.Compilers\1.1.0-rc1-20151014-05\tools\Microsoft.CSharp.Core.targets(103,11): error MSB4064: The "Deterministic" parameter is not supported by the "Csc" task. Verify the parameter exists on the task, and it is a settable public instance property. [d:\j\workspace\roslyn_prtest_win_dbg_unit64\src\Tools\Source\CompilerGeneratorTools\Source\CSharpSyntaxGenerator\CSharpSyntaxGenerator.csproj]
20:49:51 C:\Users\dotnet-bot\.nuget\packages\Microsoft.Net.Compilers\1.1.0-rc1-20151014-05\tools\Microsoft.CSharp.Core.targets(67,5): error MSB4063: The "Csc" task could not be initialized with its input parameters.  [d:\j\workspace\roslyn_prtest_win_dbg_unit64\src\Tools\Source\CompilerGeneratorTools\Source\CSharpSyntaxGenerator\CSharpSyntaxGenerator.csproj]
20:49:51 C:\Users\dotnet-bot\.nuget\packages\Microsoft.Net.Compilers\1.1.0-rc1-20151014-05\tools\Microsoft.CSharp.Core.targets(103,11): error MSB4064: The "Deterministic" parameter is not supported by the "Csc" task. Verify the parameter exists on the task, and it is a settable public instance property. [d:\j\workspace\roslyn_prtest_win_dbg_unit64\src\Tools\Source\CompilerGeneratorTools\Source\CSharpErrorFactsGenerator\CSharpErrorFactsGenerator.csproj]
20:49:51 C:\Users\dotnet-bot\.nuget\packages\Microsoft.Net.Compilers\1.1.0-rc1-20151014-05\tools\Microsoft.CSharp.Core.targets(67,5): error MSB4063: The "Csc" task could not be initialized with its input parameters.  [d:\j\workspace\roslyn_prtest_win_dbg_unit64\src\Tools\Source\CompilerGeneratorTools\Source\CSharpErrorFactsGenerator\CSharpErrorFactsGenerator.csproj]
20:49:51 C:\Users\dotnet-bot\.nuget\packages\Microsoft.Net.Compilers\1.1.0-rc1-20151014-05\tools\Microsoft.VisualBasic.Core.targets(90,11): error MSB4064: The "Deterministic" parameter is not supported by the "Vbc" task. Verify the parameter exists on the task, and it is a settable public instance property. [d:\j\workspace\roslyn_prtest_win_dbg_unit64\src\Tools\Source\CompilerGeneratorTools\Source\VisualBasicSyntaxGenerator\VisualBasicSyntaxGenerator.vbproj]
20:49:51 C:\Users\dotnet-bot\.nuget\packages\Microsoft.Net.Compilers\1.1.0-rc1-20151014-05\tools\Microsoft.VisualBasic.Core.targets(56,5): error MSB4063: The "Vbc" task could not be initialized with its input parameters.  [d:\j\workspace\roslyn_prtest_win_dbg_unit64\src\Tools\Source\CompilerGeneratorTools\Source\VisualBasicSyntaxGenerator\VisualBasicSyntaxGenerator.vbproj]
20:49:51 C:\Users\dotnet-bot\.nuget\packages\Microsoft.Net.Compilers\1.1.0-rc1-20151014-05\tools\Microsoft.VisualBasic.Core.targets(90,11): error MSB4064: The "Deterministic" parameter is not supported by the "Vbc" task. Verify the parameter exists on the task, and it is a settable public instance property. [d:\j\workspace\roslyn_prtest_win_dbg_unit64\src\Tools\Source\CompilerGeneratorTools\Source\VisualBasicErrorFactsGenerator\VisualBasicErrorFactsGenerator.vbproj]
20:49:51 C:\Users\dotnet-bot\.nuget\packages\Microsoft.Net.Compilers\1.1.0-rc1-20151014-05\tools\Microsoft.VisualBasic.Core.targets(56,5): error MSB4063: The "Vbc" task could not be initialized with its input parameters.  [d:\j\workspace\roslyn_prtest_win_dbg_unit64\src\Tools\Source\CompilerGeneratorTools\Source\VisualBasicErrorFactsGenerator\VisualBasicErrorFactsGenerator.vbproj]
20:49:51 C:\Users\dotnet-bot\.nuget\packages\Microsoft.Net.Compilers\1.1.0-rc1-20151014-05\tools\Microsoft.CSharp.Core.targets(103,11): error MSB4064: The "Deterministic" parameter is not supported by the "Csc" task. Verify the parameter exists on the task, and it is a settable public instance property. [d:\j\workspace\roslyn_prtest_win_dbg_unit64\src\Debugging\Microsoft.DiaSymReader.PortablePdb\Microsoft.DiaSymReader.PortablePdb.csproj]
20:49:51 C:\Users\dotnet-bot\.nuget\packages\Microsoft.Net.Compilers\1.1.0-rc1-20151014-05\tools\Microsoft.CSharp.Core.targets(67,5): error MSB4063: The "Csc" task could not be initialized with its input parameters.  [d:\j\workspace\roslyn_prtest_win_dbg_unit64\src\Debugging\Microsoft.DiaSymReader.PortablePdb\Microsoft.DiaSymReader.PortablePdb.csproj]
20:49:59   CompilersBoundTreeGenerator -> d:\j\workspace\roslyn_prtest_win_dbg_unit64\Binaries\Debug\BoundTreeGenerator.exe
20:50:03   CodeAnalysis -> d:\j\workspace\roslyn_prtest_win_dbg_unit64\Binaries\Debug\Microsoft.CodeAnalysis.dll
```

**Comment by amcasey** [comment]
Affected machines: dci-win-fbld-17

**Comment by amcasey** [comment]
Unable to repro the issue on dci-win-fbld-17, so deleted the workspaces of runs that had failed and rebooted.  Waiting for new repro.

**Comment by davkean** [comment]
The behavior here, could be due to MSBuild node reuse, so we're picking up the older version of the task.

**Comment by davkean** [comment]
We hit this again: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/441/consoleFull#32055789679494335-f7bd-47d0-8771-8661e00c2db2

**Comment by TyOverby** [comment]
And again:

> http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/570/consoleFull#32055789679494335-f7bd-47d0-8771-8661e00c2db2 failed with:

> C:\Users\dotnet-bot\.nuget\packages\Microsoft.Net.Compilers\1.1.0-rc1-20151014-05\tools\Microsoft.CSharp.Core.targets(103,11): error MSB4064: The "Deterministic" parameter is not supported by the "Csc" task. Verify the parameter exists on the task, and it is a settable public instance property. [d:\j\workspace\roslyn_prtest_win_dbg_unit32\src\Tools\Source\CompilerGeneratorTools\Source\CSharpSyntaxGenerator\CSharpSyntaxGenerator.csproj]

**Comment by davkean** [comment]
Looks like we might still be hitting this: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit64/612/. It looks the merge included my change, unless MSBuild happened to pick up some pre __existing__ nodes from before my change. We should watch over the next couple of days to see if we're still hitting this.

**Comment by davkean** [comment]
It's extremely clear that we haven't fixed this, perhaps we're picking up nodes even though reuse is false? I thought I saw that reuseNodes only shut down nodes at the end of the build, but not actually prevent us from picking up existing nodes - but I could be wrong.

**Comment by jaredpar** [comment]
@davkean can you link to some recent failures of this nature?

**Comment by amcasey** [comment]
http://dotnet-ci.cloudapp.net/view/roslyn/job/roslyn_master_win_dbg_unit32/249/

**Comment by amcasey** [comment]
http://dotnet-ci.cloudapp.net/view/roslyn/job/roslyn_master_win_rel_unit64/246/

**Comment by amcasey** [comment]
I'm not sure how to efficiently search Jenkins, but those are the most recent I can find.

**Comment by jaredpar** [comment]
@amcasey yep that is pretty definitive now.  Here are the events:

- Node reuse is disabled October 27 via d8371c75dec963fd92eb93917005d426a3ce256d 
- Jenkins run 249 for master_win_dbg used commit 88612f48534512b2cc09b3f61553c1753aa0b644 from October 28.

That run failed with the following output:

``` 
22:36:55 d:\j\workspace\roslyn_master_win_dbg_unit32>msbuild /nologo /v:m /m /nodeReuse:false /p:BuildVersion=0.0.0.0 d:\j\workspace\roslyn_master_win_dbg_unit32\build/Toolset.sln /p:NuGetRestorePackages=false /p:Configuration=Debug 
22:36:59 
C:\Users\dotnet-bot\.nuget\packages\Microsoft.Net.Compilers\1.1.0-rc1-20151014-05\tools\Microsoft.CSharp.Core.targets(103,11): error MSB4064: The "Deterministic" parameter is not supported by the "Csc" task. Verify the parameter exists on the task, and it is a settable public instance property. [d:\j\workspace\roslyn_master_win_dbg_unit32\src\Tools\Source\CompilerGeneratorTools\Source\CSharpSyntaxGenerator\CSharpSyntaxGenerator.csproj]
```

It's very clear that the noderuse is passed as false and the error still occurs. 

At this point I think we need to get more involvement from the MSBuild team.  Clearly some level of DLL reuse is still occuring in MSBuild even though we specifically asked for it to be disabled.

**Comment by davkean** [comment]
Okay, I've got further on this, this looks to be an MSBuild bug, if you do the following:

1. Run `msbuild BuildAndTest.proj /m /nodereuse:true /t:build`
2. Wait for the build to finish and notice that there's as many MSBuild processes left open as there are logical CPUs - 1. 
3. Run `msbuild BuildAndTest.proj /m /nodereuse:false /t:build`
4. Notice that MSBuild only spawns one more process, it's basically reusing the existing processes.

This means that if the code analysis task has already been loaded, we will run in the above issue.

Basically, all it takes is one build to reuse nodes, and all subsequent builds could pick up the left over processes.

**Comment by jaredpar** [comment]
So basically nodeReuse dictates what the next guy does ...

**Comment by rainersigwald** [comment]
I can confirm that that's the "expected" node-reuse behavior of MSBuild.  Filed https://github.com/Microsoft/msbuild/issues/339 to fix that.

For a CI build, if you own the machine you could kill MSBuild.exe before starting your build (I know this is horrible).  Obviously that's more dangerous on a machine that might be doing another build (including in VS).

I'm a bit surprised that this is manifesting in the way we're seeing here, though.  How are you loading the tasks?  If they're just-built and have the same version number, maybe our task loader is optimizing away the load.

**Comment by davkean** [comment]
I'm going to close this, while we didn't fix the underying issue, we stopped running parallel jobs - which I figured out is the only time this would occur. Jenkins already shuts down processes at the end of the job.

**Comment by rainersigwald** [comment]
And just so it's documented here, the MSBuild-side fix to avoid this problem everywhere will ship in a future Visual Studio update (it's already checked in to the appropriate branch, but too late for 2015 Update 1).
